library/aws/policies_v2.ts

```diff
@@ -0,0 +1,326 @@
+import * as AR from "../../providers/aws/resources.ts";
+import { ArnSecret } from "./secrets.ts";
+
+export interface NamedPolicy {
+  name: string;
+  policy: {
+    Statement: Statement[],
+    [key: string]: unknown;
+  };
+}
+
+export interface Statement {
+  Action: unknown;
+  Effect: unknown;
+  [key: string]: unknown;
+}
+
+export function newNamedPolicy(
+  name: string,
+  statements: Statement[],
+): NamedPolicy {
+  return {
+    name,
+    policy: {
+      Version: "2012-10-17",
+      Statement: statements,
+    },
+  };
+}
+
+export function combineNamedPolicies(
+  name: string,
+  namedPolicies: NamedPolicy[],
+): NamedPolicy {
+  const Statement: Statement[] = [];
+  for (const np of namedPolicies) {
```

Comment:
This is 
```
const Statement: Statement[] = namedPolicies.map((np) => np.policy.Statement);
```

(You can ignore this comment if you prefer for loop)

Reply:
Not a strong preference, though I actually wrote the map before I rewrote as a loop. I'll change it back!

Reply:
Actually, I remember now why I wrote the loop. You actually need something like:

```
const Statement: Statement[] = [].concat(...namedPolicies.map((np) => np.policy.Statement)));
```

which seemed ugly.